test/e2e/conformance_test.go

```diff
@@ -67,6 +68,14 @@ var _ = Describe("Conformance Tests", func() {
 	Measure(specName, func(b Benchmarker) {
 		var err error
 
+		kubernetesVersion := e2eConfig.GetVariable(capi_e2e.KubernetesVersion)
+		flavor := clusterctl.DefaultFlavor
+		if useCIArtifacts {
+			flavor = "conformance-ci-artifacts"
+			kubernetesVersion, err = kubernetesversions.LatestCIRelease()
+			Expect(os.Setenv("CI_VERSION", kubernetesVersion)).NotTo(HaveOccurred())
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
Maybe these two lines should be swapped? That is, don't bother calling `Setenv` if there was already an error.